File: java/example-supplier/src/main/java/dev/kaiquebt/anycall/example/supplier/SentimentAnalyzer.java
package dev.kaiquebt.anycall.example.supplier;

import dev.kaiquebt.anycall.annotation.Supply;
import dev.kaiquebt.anycall.example.model.Sentiment;
import dev.kaiquebt.anycall.example.model.TextRequest;

public class SentimentAnalyzer {

    @Supply("analyze-sentiment")
    public Sentiment analyzeSentiment(TextRequest req) {
        return new Sentiment(req.getText(), "positive");
    }
}

File: implementations/python/example-supplier/src/example_supplier/sentiment_analyzer.py
from anycall import supply

from .model.text_request import TextRequest
from .model.sentiment import Sentiment


class SentimentAnalyzer:
    """Analyzer for sentiment analysis operations."""

    @supply("analyze-sentiment")
    def analyze_sentiment(self, req: TextRequest) -> Sentiment:
        """Analyze sentiment of text.

        Args:
            req: Text request

        Returns:
            Sentiment analysis result
        """
        return Sentiment(text=req.text, label="positive")
